pkg/aws/eni/instances.go

```diff
@@ -204,6 +205,11 @@ func (m *InstancesManager) resync(ctx context.Context, instanceID string) time.T
 		return time.Time{}
 	}
 
+	err = m.limitsGetter.InitEC2APIUpdateTrigger(m.api, limits.TriggerMinInterval, limits.EC2apiTimeout, limits.EC2apiRetryCount)
```

Comment:
I find it weird that we are calling this function here periodically, even though we could all of this in `limits.NewLimitsGetter()` instead (since the function itself does a `sync.Once` after all).

I understand making `NewInstancesManager` return an error causes a bit of churn in the unit tests, but I think it would be cleaner overall to merge `InitEC2APIUpdateTrigger` into `NewLimitsGetter`.

Reply:
I thought it's better to seperate the logic from init and start it. All the others ec2 api happenning here and that's why I put it.